lib/fizzy/execute.cpp

```diff
@@ -415,24 +415,22 @@ inline bool store_into_memory(bytes& memory, Stack<uint64_t>& stack, const uint8
 }
 
 template <typename Op>
-inline void unary_op(Stack<uint64_t>& stack, Op op) noexcept
+inline void unary_op(OperandStack& stack, Op op) noexcept
 {
-    using T = decltype(op(stack.pop()));
-    const auto a = static_cast<T>(stack.pop());
-    stack.push(op(a));
+    using T = decltype(op(stack.top()));
+    stack.top() = op(static_cast<T>(stack.top()));
 }
 
 template <typename Op>
-inline void binary_op(Stack<uint64_t>& stack, Op op) noexcept
+inline void binary_op(OperandStack& stack, Op op) noexcept
 {
-    using T = decltype(op(stack.pop(), stack.pop()));
+    using T = decltype(op(stack.top(), stack.top()));
     const auto val2 = static_cast<T>(stack.pop());
-    const auto val1 = static_cast<T>(stack.pop());
-    stack.push(static_cast<std::make_unsigned_t<T>>(op(val1, val2)));
+    stack.top() = static_cast<std::make_unsigned_t<T>>(op(static_cast<T>(stack.top()), val2));
```

Comment:
I'd keep `val1`, this line looks too complicated (should be easy for compiler to optimize it away)
```suggestion
    const auto val1 = static_cast<T>(stack.top());
    stack.top() = static_cast<std::make_unsigned_t<T>>(op(val1, val2));
```
Or as alternative you could at least declare `T_unsigned` on a separate line